src/com/trilead/ssh2/crypto/keys/Ed25519PrivateKey.java

```diff
@@ -0,0 +1,137 @@
+package com.trilead.ssh2.crypto.keys;
+
+import com.trilead.ssh2.packets.TypesReader;
+import com.trilead.ssh2.packets.TypesWriter;
+
+import java.io.IOException;
+import java.security.PrivateKey;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.util.Arrays;
+
+import javax.security.auth.DestroyFailedException;
+
+public class Ed25519PrivateKey implements PrivateKey {
```

Comment:
This class is already provided by Bouncy Castle, the EdDSA library, and even Java itself starting around version 14 or so. Is there a need for a wrapper class here if you use the classes from the crypto library?

Reply:
@jvz so I it exists in this library:
<dependency>
      <groupId>net.i2p.crypto</groupId>
      <artifactId>eddsa</artifactId>
      <version>0.3.0</version>
    </dependency>

Reply:
Based on some external exploration I did in this area, using the i2p library class is probably the easiest way to do so. BouncyCastle has a Java 15 provider for the new EdDSA APIs added there, but that's not super useful for Java 8-14. If this class is reintroduced in a subsequent PR, it's fine as the API around XDH and EdDSA in Java are awkward being introduced in Java 11 and 15 respectively.